pages/api/search/index.ts

```diff
@@ -17,23 +18,25 @@ export default async function handler(
     : "current";
 
   try {
-    const { results } = await client.multipleQueries(
-      getSearchQueries(req.query.query as string, docsVer),
-      {
-        strategy: "stopIfEnoughMatches",
-      }
-    );
-    rawResults = results;
+    const { hits } = await index.search(req.query.query as string, {
+      hitsPerPage: 50,
+      filters: `docs_ver:${docsVer}`,
+      restrictHighlightAndSnippetArrays: true,
+      attributesToHighlight: ["content", "headers"],
+      attributesToSnippet: ["content:30", "headers"],
+      snippetEllipsisText: "…",
+      highlightPreTag: "<b class='found-part'>",
+      highlightPostTag: "</b>",
+    });
 
-    const finishResult = rawResults.reduce((total, res) => {
-      total.push(...res.hits);
-      total.map((hit) => {
-        hit.title = hit.title.includes("|")
-          ? hit.title.split("|")[0]
-          : hit.title;
-      });
-      return total;
-    }, []);
+    rawResults = hits;
+
+    const finishResult = rawResults.map((hit) => {
+      const title = hit.title.includes("|")
+        ? hit.title.split("|")[0].trim()
+        : hit.title;
```

Comment:
When calling `hit.title.includes` you need to ensure title exists, or you could get an undefined error:
```ts
const title = hit.title?.includes("|")
        ? hit.title.split("|")[0].trim()
        : hit.title;
```